community/kernel/src/main/java/org/neo4j/kernel/impl/locking/community/RWLock.java

```diff
@@ -545,6 +496,39 @@ public String toString()
         return "RWLock[" + resource + ", hash="+hashCode()+"]";
     }
 
+    private void acquireLock( LockType type, Object tx, TxTermination termination, TxLockElement tle )
+    {
+        ragManager.checkWaitOn( this, tx );
+        WaitElement waitElement = new WaitElement( tle, type, currentThread() );
```

Comment:
it looks like now when you will be interrupted and will try to acquireLock N times you will add WaitElement N times, without calling `ragManager.stopWaitOn( this, tx )`, so wait elements will be accumulated ?